sklearn/cluster/tests/test_optics.py

```diff
@@ -780,14 +790,14 @@ def test_extract_dbscan():
     C2 = [4, -1] + 0.2 * rng.randn(n_points_per_cluster, 2)
     C3 = [1, 2] + 0.2 * rng.randn(n_points_per_cluster, 2)
     C4 = [-2, 3] + 0.2 * rng.randn(n_points_per_cluster, 2)
-    X = np.vstack((C1, C2, C3, C4))
+    X = np.vstack((C1, C2, C3, C4)).astype(global_dtype, copy=False)
 
     clust = OPTICS(cluster_method="dbscan", eps=0.5).fit(X)
     assert_array_equal(np.sort(np.unique(clust.labels_)), [0, 1, 2, 3])
 
 
-def test_precomputed_dists():
-    redX = X[::2]
+def test_precomputed_dists(global_dtype):
+    redX = X[::2].astype(global_dtype, copy=False)
```

Comment:
A few lines later comes an `assert_allclose`. According to #22881, this should be replaced by `sklearn.utils._testing.assert_allclose`, right?

Reply:
The instance of `assert_allclose` is already one of `sklearn.utils._testing.assert_allclose` here.

Reply:
The we can forget my comment and merge.